src/reporters/OptimizationData.C

```diff
@@ -61,6 +77,28 @@ OptimizationData::OptimizationData(const InputParameters & parameters)
 void
 OptimizationData::execute()
 {
+  if (!_var)
```

Comment:
why do you use this _var instead of the measurementValuesReporter?  Is this executed on the forward problem?